src/module/item/consumable/index.ts

```diff
@@ -190,12 +299,30 @@ class ConsumablePF2e extends PhysicalItemPF2e {
             return spellcastingEntries[bestEntry];
         })();
 
-        if (entry) {
-            const systemData = spell.system;
-            if (entry instanceof SpellcastingEntryPF2e) {
-                systemData.location.value = entry.id;
+        return entry ? entry : null;
+    }
+
+    async castEmbeddedSpell(
+        this: Embedded<ConsumablePF2e>,
+        spell?: Embedded<SpellPF2e>,
+        trickMagicItemData?: TrickMagicItemEntry
+    ): Promise<void> {
```

Comment:
I think you can omit the spell parameter since embedded spell is cache. There's no longer a performance gain from doing this and it'll make the API less awkward.

Reply:
I'm only caching spells from embedded data and from a compendium currently. Caching world items has the problem that changes to the source items are not reflected until the consumable data is prepared again.